src/PostageStamp.sol

```diff
@@ -110,6 +84,43 @@ contract PostageStamp is AccessControl, Pausable {
     // Normalised balance at the blockheight expire() was last called.
     uint256 public lastExpiryBalance;
 
+    // ----------------------------- Events ------------------------------
+
+    /**
+     * @dev Emitted when a new batch is created.
+     */
+    event BatchCreated(
+        bytes32 indexed batchId,
+        uint256 totalAmount,
+        uint256 normalisedBalance,
+        address owner,
+        uint8 depth,
+        uint8 bucketDepth,
+        bool immutableFlag
+    );
+
+    /**
+     * @dev Emitted when an pot is Withdrawn.
+     */
+    event PotWithdrawn(address recipient, uint256 totalAmount);
+
+    /**
+     * @dev Emitted when an existing batch is topped up.
+     */
+    event BatchTopUp(bytes32 indexed batchId, uint256 topupAmount, uint256 normalisedBalance);
+
+    /**
+     * @dev Emitted when the depth of an existing batch increases.
+     */
+    event BatchDepthIncrease(bytes32 indexed batchId, uint8 newDepth, uint256 normalisedBalance);
+
+    /**
+     *@dev Emitted on every price update.
+     */
+    event PriceUpdate(uint256 price);
+
+    // ----------------------------- CONSTRUCTOR ------------------------------
```

Comment:
I would remove this, it does not add much to anything, on the other hand introduces to obligation to write this above every constructor.

Reply:
ahh, well its also a convention to use it. good solidity code will always have it. not really much overhead as you have one constructor per contract, so not much extra work, but always nice to have when scrolling as it is expected to be there if all all sections are commented.